airbyte-webapp/src/components/connectorBuilder/Builder/PaginationSection.tsx

```diff
@@ -0,0 +1,180 @@
+import { useField } from "formik";
+
+import GroupControls from "components/GroupControls";
+import { ControlLabels } from "components/LabeledControl";
+
+import { injectIntoValues } from "../types";
+import { BuilderCard } from "./BuilderCard";
+import { BuilderField } from "./BuilderField";
+import { BuilderOneOf } from "./BuilderOneOf";
+
+interface PaginationSectionProps {
+  streamFieldPath: (fieldPath: string) => string;
+}
+
+export const PaginationSection: React.FC<PaginationSectionProps> = ({ streamFieldPath }) => {
+  const [field, , helpers] = useField(streamFieldPath("paginator"));
+
+  const handleToggle = (newToggleValue: boolean) => {
+    if (newToggleValue) {
+      helpers.setValue({
+        strategy: {
+          type: "OffsetIncrement",
+        },
+        pageSizeOption: {
+          inject_into: "request_parameter",
+        },
+        pageTokenOption: {
+          inject_into: "request_parameter",
+        },
+      });
+    } else {
+      helpers.setValue(undefined);
+    }
+  };
+  const toggledOn = field.value !== undefined;
+
+  const pageTokenOption = (
+    <GroupControls
+      label={
+        <ControlLabels
+          label="Page token option"
+          infoTooltipContent="Configures how the page token will be sent in requests to the source API"
+        />
+      }
+    >
+      <BuilderField
+        type="enum"
+        path={streamFieldPath("paginator.pageTokenOption.inject_into")}
+        options={injectIntoValues}
+        label="Inject into"
+        tooltip="Configures where the page token should be set on the HTTP requests"
+      />
+      <BuilderField
```

Comment:
This seems to be too simplistic - it depends on the inject_into whether or not you are allowed to provide a field name:
<img width="1054" alt="Screenshot 2022-12-21 at 14 34 38" src="https://user-images.githubusercontent.com/1508364/208919355-f02fbf17-fd4d-43a5-aaed-e551a7d9519e.png">
<img width="1034" alt="Screenshot 2022-12-21 at 14 43 12" src="https://user-images.githubusercontent.com/1508364/208919360-84814167-6be6-445b-bc72-46daf1e4ca58.png">